api/validation/config.go

```diff
@@ -0,0 +1,22 @@
+package validation
+
+import (
+	"errors"
+
+	apps "github.com/ninech/apis/apps/v1alpha1"
+)
+
+// ConfigValidator validates a Config
+type ConfigValidator struct {
+	Config apps.Config
+}
+
+// Validate validates the config
+func (c ConfigValidator) Validate() error {
```

Comment:
Should we also copy over the other validations we have on the backend for the config (e.g env vars)?

Reply:
As discussed, it's fine if we merge this as is and add more validation later. We anyways have validation on the API side so it will just fall back to that.